ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/AutoMediaBrowserService.kt

```diff
@@ -0,0 +1,186 @@
+package org.moire.ultrasonic.service
+
+import android.os.Bundle
+import android.os.Handler
+import android.support.v4.media.MediaBrowserCompat
+import android.support.v4.media.MediaDescriptionCompat
+import android.support.v4.media.session.MediaSessionCompat
+import androidx.media.MediaBrowserServiceCompat
+import androidx.media.utils.MediaConstants
+import org.koin.android.ext.android.inject
+import org.moire.ultrasonic.util.MediaSessionEventDistributor
+import org.moire.ultrasonic.util.MediaSessionEventListener
+import timber.log.Timber
+
+
+const val MY_MEDIA_ROOT_ID = "MY_MEDIA_ROOT_ID"
+const val MY_MEDIA_ALBUM_ID = "MY_MEDIA_ALBUM_ID"
+const val MY_MEDIA_ARTIST_ID = "MY_MEDIA_ARTIST_ID"
+const val MY_MEDIA_ALBUM_ITEM = "MY_MEDIA_ALBUM_ITEM"
+const val MY_MEDIA_LIBRARY_ID = "MY_MEDIA_LIBRARY_ID"
+const val MY_MEDIA_PLAYLIST_ID = "MY_MEDIA_PLAYLIST_ID"
+
+class AutoMediaBrowserService : MediaBrowserServiceCompat() {
+
+    private lateinit var mediaSessionEventListener: MediaSessionEventListener
+    private val mediaSessionEventDistributor: MediaSessionEventDistributor by inject()
+    private val lifecycleSupport: MediaPlayerLifecycleSupport by inject()
+
+    override fun onCreate() {
+        super.onCreate()
+
+        mediaSessionEventListener = object : MediaSessionEventListener {
+            override fun onMediaSessionTokenCreated(token: MediaSessionCompat.Token) {
+                Timber.i("AutoMediaBrowserService onMediaSessionTokenCreated called")
+                if (sessionToken == null) {
+                    Timber.i("AutoMediaBrowserService onMediaSessionTokenCreated session token was null, set it to %s", token.toString())
+                    sessionToken = token
+                }
+            }
+
+            override fun onPlayFromMediaIdRequested(mediaId: String?, extras: Bundle?) {
+                // TODO implement
```

Comment:
This will be called from the MediaSession.
Currently I imagine this should add the selected item(s) to the playlist as next track(s) and start playing it. 
This is debatable, but if the user has already loaded a playlist, I'd just add the selected tracks to it without removing anything. For me the main use case in the car is that some music always plays in the background, like a radio - if I select some songs directly, I want to listen to those, and then get back to radio mode :)